### PR TITLE
build: configure GitHub Actions

### PR DESCRIPTION
**Pull request by kyubisation**

### CODE REVIEW

**Comment by tongpu** [comment]
@kyubisation Any ideas why the actions runner is unable to push the built image?

**Comment by kyubisation** [comment]
Unfortunately no. But there is updated documentation on how to authenticate: https://docs.github.com/en/packages/managing-github-packages-using-github-actions-workflows/publishing-and-installing-a-package-with-github-actions#upgrading-a-workflow-that-accesses-a-registry-using-a-personal-access-token

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -23,3 +26,17 @@ jobs:
           cache: 'maven'
       - name: Run the Maven verify phase
         run: mvn --batch-mode --update-snapshots verify
+
+      # For pull requests we create a container image and if it is not from a fork, we push it to the
+      # GitHub Container Registry ghcr.io as a preview image with the tag name 'pr{pull request number}'.
+      - name: Login to GitHub Container Registry
+        if: github.event_name == 'pull_request' && !github.event.pull_request.head.repo.fork
+        run: echo ${{ secrets.GITHUB_TOKEN }} | docker login ghcr.io --username ${{ github.actor }} --password-stdin
```
Comment (kyubisation) [inline .github/workflows/continuous-integration.yml:34]:
```suggestion
        run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io --username $ --password-stdin
```

**Comment by kyubisation** [comment]
It has been a while since I had to do this, so I might have forgotten details. If my fix does not work, we can schedule a debug session.

**Comment by aiAdrian** [comment]
I don't have a detailed understanding of what this PR is for and what the changes trigger. So please take a look at Lukas's PR, thanks.

**Comment by kyubisation** [comment]
Alright, I fixed the issue with pushing. The problem ist that the maven process does not have access to the container registry credentials and fails, if it attempts to push.
I separated the push into its own command and now it works.

**Review by tongpu — APPROVED** [review]

**Comment by kyubisation** [comment]
No problem 👍

**Pull request merged**